ADR/adr-0010-namespace-metadata

```diff
@@ -0,0 +1,54 @@
+# ADR-0010 Namespace Metadata
+
+Date: 2022-December-9
+
+## Status
+
+Accepted
+
+## Approvers
+
+* Ann Marie Fred
+* Gorkem Ercan
+
+## Reviewers
+
+* Matous Jobanek
+* Ralph Bean
+
+## Context
+
+We need metadata on our namespaces to make Stonesoup easier to operate and maintain. Standardizing namespace metadata will make it easier for us to search our logs and metrics across clusters. It will also allow us to enable logging for outgoing network traffic, one of our security requirements.
+
+## Namespace labels
+
+We will apply the following labels to Stonesoup namespaces, to make them easier to identify programmatically:
+
+- `appstudio.redhat.com/namespacetype: "controller"` for namespaces containing controllers developed for Stonesoup. For example, we would annotate the `gitops-service-argocd` namespace but not the `openshift-gitops` namespace.
+- `appstudio.redhat.com/namespacetype: "user-workspace-data"` for User workspaces where Applications, Components, and so on are stored
+- `appstudio.redhat.com/namespacetype: "user-deployments"` for the namespaces where GitOps deploys applications for users
+- `appstudio.redhat.com/namespacetype: "user-builds"` for the namespaces where the Pipeline Service manages users' PipelineRun resources
```

Comment:
Let's explicitly mention that one namespace can have multiple types.

Reply:
Added.  I will ask @hmariset to come back and add a little more context on how the billing team will use the labels.  I assume they will be the ones to apply them, or that they will contribute the code to do so.

Reply:
stonesoup-o11y team will copy these labels and attach them to all the data we aggregate for billing.
The SKUs data probably needs to be propagated from the UI during the user onboarding.

Reply:
OK let's assume that actually populating the labels will have its own feature and epics, as opposed to designing it on the fly here.